### DISCUSSION TITLE
can we raise custom error message when a constraint fails ?

### DISCUSSION THREAD

**Discussion by subramanian0331** [cue-lang/cue#2199] [Q&A]
`count: >5   @error( server count should be >5 )`

can we create custom error to cue constraints?

**Comment by verdverm** [comment]
Not currently, unless you use the GoAPI to parse it out and present it. I think there is an open issue around supporting this, but the design space is a little more complicated.

consider:

```
port: int @error(port must be set)

port: >1024 @error(port must not be privileged)
```

(https://cuelang.org/play/?id=k5YIOKjxMhm#cue@export@cue)

Which error should be presented here?

Attributes are also stripped through references, so the error message can be lost

```
prod: count: >5   @error( server count should be >5 )

fleet: {
  count: prod.count
}
```

(https://cuelang.org/play/?id=wANF_9YPb3a#cue@export@cue)

This one is intentional because of

```
#A: {
	x: int @protobuf(1,int64)
	y: int @protobuf(2,int64)
}

a: {
	x: 1
	y: x // propagation issue here
}

-- output --

// would result in a conflict for indexes
a: {
	y: 1 @protobuf(1,int64) @protobuf(2,int64)
}
```
